// ews-javascript-api/lib/EwsOAuthHelper.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __classPrivateFieldGet = (this && this.__classPrivateFieldGet) || function (receiver, state, kind, f) {
    if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a getter");
    if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot read private member from an object whose class did not declare it");
    return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
};
var __classPrivateFieldSet = (this && this.__classPrivateFieldSet) || function (receiver, state, value, kind, f) {
    if (kind === "m") throw new TypeError("Private method is not writable");
    if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a setter");
    if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot write private member to an object whose class did not declare it");
    return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
};
var _EwsOAuthHelper_tokenObject, _EwsOAuthHelper_cca, _EwsOAuthHelper_scopes, _EwsOAuthHelper_loginUrl;
Object.defineProperty(exports, "__esModule", { value: true });
exports.EwsOAuthHelper = void 0;
const msal_node_1 = require("@azure/msal-node");
class EwsOAuthHelper {
    get cachedToken() {
        return __classPrivateFieldGet(this, _EwsOAuthHelper_tokenObject, "f");
    }
    constructor({ clientId, clientSecret, tenantId, scopes, loginUrl }) {
        _EwsOAuthHelper_tokenObject.set(this, void 0);
        _EwsOAuthHelper_cca.set(this, void 0);
        _EwsOAuthHelper_scopes.set(this, ['https://outlook.office365.com/.default']);
        _EwsOAuthHelper_loginUrl.set(this, 'https://login.microsoftonline.com/');
        if (scopes)
            __classPrivateFieldSet(this, _EwsOAuthHelper_scopes, [...scopes], "f");
        if (loginUrl)
            __classPrivateFieldSet(this, _EwsOAuthHelper_loginUrl, loginUrl, "f");
        __classPrivateFieldSet(this, _EwsOAuthHelper_cca, new msal_node_1.ConfidentialClientApplication({ auth: { clientId, clientSecret, authority: `${__classPrivateFieldGet(this, _EwsOAuthHelper_loginUrl, "f")}${tenantId}` } }), "f");
    }
    getAppAccessToken(_a) {
        return __awaiter(this, arguments, void 0, function* ({ cache = true, raw = false }) {
            const result = yield __classPrivateFieldGet(this, _EwsOAuthHelper_cca, "f").acquireTokenByClientCredential({
                scopes: __classPrivateFieldGet(this, _EwsOAuthHelper_scopes, "f"),
                skipCache: !cache,
            });
            __classPrivateFieldSet(this, _EwsOAuthHelper_tokenObject, result, "f");
            if (raw)
                return result;
            return result.accessToken;
        });
    }
}
exports.EwsOAuthHelper = EwsOAuthHelper;
_EwsOAuthHelper_tokenObject = new WeakMap(), _EwsOAuthHelper_cca = new WeakMap(), _EwsOAuthHelper_scopes = new WeakMap(), _EwsOAuthHelper_loginUrl = new WeakMap();
